tools/xtce_generation/telemetry_generator.py

```diff
@@ -1,16 +1,20 @@
 import xml.etree.ElementTree as Et
 from datetime import datetime
 
+import pandas as pd
+
 from tools.xtce_generation.ccsds_header_xtce_generator import CCSDSParameters
 
 
 class TelemetryGenerator:
     def __init__(self, packet_name, path_to_excel_file, apid, sci_byte=None, pkt=None):
         self.packet_name = packet_name
-        self.path_to_excel_file = path_to_excel_file
         self.apid = apid
         self.sci_byte = sci_byte
-        self.pkt = pkt
+
+        # Read excel sheet
+        xls = pd.ExcelFile(path_to_excel_file)
+        self.pkt = xls.parse(packet_name)
```

Comment:
@maxinelasp If I change it to this, will it help?
```suggestion
         if pkt is None:
            # Read excel sheet
            xls = pd.ExcelFile(path_to_excel_file)
            self.pkt = xls.parse(packet_name)
        else:
            self.pkt = pkt
```

Reply:
Yes!

Reply:
cool. I will make that change.